izpack-installer/src/main/java/com/izforge/izpack/util/os/Unix_Shortcut.java

```diff
@@ -493,36 +531,67 @@ public void save() throws Exception
             // Now install my Own with xdg-if available // Note the The reverse Uninstall-Task is on
             // TODO: "WHICH another place"
 
+            UnixUser sudoUser = getSudoUser();
+            
             String cmd = getXdgDesktopIconCmd();
             if (cmd != null)
             {
                 createExtXdgDesktopIconCmd(shortCutLocation);
-                // / TODO: DELETE the ScriptFiles
-                myInstallScript.appendln(new String[]{
+                
+                String[] installCmd = new String[]{
                     makeFilenameScriptable(myXdgDesktopIconCmd),
                     "install",
                     "--novendor",
-                    StringTool.escapeSpaces(writtenDesktopFile.toString())});
-                ShellScript myUninstallScript = new ShellScript();
-                myUninstallScript.appendln(new String[]{
+                    StringTool.escapeSpaces(writtenDesktopFile.toString())};
+                String[] uninstallCmd = new String[]{
                     makeFilenameScriptable(myXdgDesktopIconCmd),
                     "uninstall",
                     "--novendor",
-                    StringTool.escapeSpaces(writtenDesktopFile.toString())});
+                    StringTool.escapeSpaces(writtenDesktopFile.toString())};
+                
+                // / TODO: DELETE the ScriptFiles
```

Comment:
Did you mean this TODO? I just moved it to a similar place since I am not the author of it and I did not inspect how this should be resolved.

Reply:
Ah sorrz, I see, it has been there before. Alright, I don't know the historical meaning of this line, iIjust hate hanging TODOs :-)